src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -183,17 +183,31 @@ private int TryGetAppSettingAsIntWithDefault(string key, int defaultValue)
 
         public object AgentRunId { get { return _serverConfiguration.AgentRunId; } }
 
+        private bool _agentEnabled;
+
+        private static string _agentEnabledAppSettingString;
+        private static bool? _agentEnabledAppSettingRead;
+        private static bool _agentEnabledAppSettingParsed;
```

Comment:
We might need to make these `volatile` since we are not using any operations with memory barriers.